biofefi/machine_learning/data.py

```diff
@@ -34,20 +35,49 @@ def __init__(
         self._normalization = normalization
         self._numerical_cols = "all"
         self._n_bootstraps = n_bootstraps
+        self._create_synthetic_data = False
+
+    class DataBuilderError(Exception):
+        """
+        Custom exception raised when data loading or
+        generation fails in DataBuilder.
+        """
+
+        pass
```

Comment:
Can we move this outside the `DataBuilder` class please?

Reply:
Ok Dan.